apps/console/src/features/organizations/components/add-organization-role-wizard.tsx

```diff
@@ -0,0 +1,470 @@
+        }
+    }, [ wizardState && wizardState[WizardStepsFormTypes.BASIC_DETAILS] ]);
+
+    /**
+     * Method to handle create role action when create role wizard finish action is triggered.
+     *
+     * @param basicData - basic data required to create role.
+     */
+    const addRole = (basicData: any): void => {
+
+        const users: CreateRoleMemberInterface[] = [];
+        const groups: CreateGroupMemberInterface[] = [];
+        const permissions: string[] = [];
+
+        if (basicData?.UserList?.length > 0) {
+            basicData?.UserList?.forEach(user => {
+                users?.push({
+                    display: user?.userName,
+                    value: user?.id
+                });
+            });
+        }
+
+        if (basicData?.GroupList?.length > 0) {
+            basicData?.GroupList?.forEach(group => {
+                groups?.push({
+                    display: group.displayName,
+                    value: group.id
+                });
+            });
+        }
+
+        if (basicData?.PermissionList?.length > 0) {
+            basicData?.PermissionList?.forEach((permission: TreeNode) => {
+                permissions?.push(permission?.key.toString());
+            });
+        }
+
+        const roleData: CreateRoleInterface = {
+            "displayName": basicData?.BasicDetails ? basicData?.BasicDetails?.roleName : basicData?.roleName,
+            "groups": groups,
+            "permissions": permissions,
+            "schemas": [
+                "urn:ietf:params:scim:schemas:extension:2.0:Role"
+            ],
+            "users": users
+        };
+
+        setIsSubmitting(true);
+
+        if (onCreateRoleRequested) {
+            onCreateRoleRequested(roleData);
+        }
+    };
+
+    /**
+     * Method to handle the create role wizard finish action.
+     *
+     */
+    const handleRoleWizardFinish = () => {
+        addRole(wizardState);
+    };
+
+    /**
+     * Generates a summary of the wizard.
+     *
+     * @return {any}
+     */
+    const generateWizardSummary = () => {
+        if (!wizardState) {
+            return;
+        }
+
+        return wizardState;
+    };
+
+    /**
+     * Handles wizard step submit.
+     *
+     * @param values - Forms values to be stored in state.
+     * @param {WizardStepsFormTypes} formType - Type of the form.
+     */
+    const handleWizardSubmit = (values: any, formType: WizardStepsFormTypes) => {
+        if (WizardStepsFormTypes.BASIC_DETAILS === formType) {
+            setSelectedUserStrore(values.domain);
+        }
+        setCurrentWizardStep(currentStep + 1);
+        setWizardState({ ...wizardState, [formType]: values });
+    };
+
+    const handleGroupUserSubmit = (value: {
+                                       [WizardStepsFormTypes.USER_LIST]: any;
+                                       [WizardStepsFormTypes.GROUP_LIST]: any;
+                                   }
+    ): void => {
```

Comment:
Fix indentation here

Reply:
Done in https://github.com/wso2/identity-apps/pull/3617/commits/630fa8dda5d13075f2d424b2493249572e255217